packages/tooling/fast-figma-plugin-msft/src/figma/node.ts

```diff
@@ -309,6 +310,52 @@ export class FigmaPluginNode extends PluginNode {
         return null;
     }
 
+    public handleManualDarkMode(): boolean {
+        if (isInstanceNode(this.node)) {
+            if (this.node.variantProperties) {
+                const currentDarkMode = this.node.variantProperties["Dark mode"];
+                if (currentDarkMode) {
+                    const color = this.getEffectiveFillColor();
+                    if (color) {
+                        let trueString = "Yes";
+                        let falseString = "No";
+                        // Thanks Figma for supporting "true", "True", "yes", and "Yes", but not doing the work to interpret it. Assume it's paired.
+                        switch (currentDarkMode) {
+                            case "yes":
+                            case "no":
+                                trueString = "yes";
+                                falseString = "no";
+                                break;
+                            case "Yes":
+                            case "No":
+                                trueString = "Yes";
+                                falseString = "No";
+                                break;
+                            case "true":
+                            case "false":
+                                trueString = "true";
+                                falseString = "false";
+                                break;
+                            case "True":
+                            case "False":
+                                trueString = "True";
+                                falseString = "False";
+                                break;
+                        }
```

Comment:
Maybe this should be broken out into it's own helper function. I can see this being useful.

It could be condensed. Here's an example using array functions.

```suggestion
const boolHelper = (str) => (isTrue) => [["No", "Yes"], ["True", "False"]]
  .find(bools => bools.find(b => b.toLowerCase() === str.toLowerCase()))
  .map(b => str.match(/^[nytf]/) ? b.toLowerCase() : b)[!isTrue ? 0 : 1];
```
Then you can just call it `"Dark mode": boolHelper(currentDarkMode)(containerIsDark)`

I have a working example with tests here: [Codepen](https://codepen.io/kungfukarl/pen/LYQobXp)

Reply:
Impressive!